src/renderer/components/stake/add/AddStake.tsx

```diff
@@ -23,8 +32,11 @@ type Props = {
   runeAsset: Asset
   assetPrice: BigNumber
   runePrice: BigNumber
-  assetBalance: BaseAmount
-  runeBalance: BaseAmount
+  assetBalance: O.Option<BaseAmount>
+  runeBalance: O.Option<BaseAmount>
```

Comment:
what is a reason to change this values to Options ? in term of this file there are 2 states showing that there is no balances now:
`O.none` and `O.some(baseAmount(0))`
this is a little bit confusing to find a difference between this states and to tell the truth i can't.  below there is a check inside of `hasAssetBalance` and the same time checks based on `O.isSome(oAssetBalance)` (same for RUNEs)

Reply:
This change was needed to check balances in a right way. Before we had just `baseAmount(0)`, even if the balances were not loaded. But this will trigger a "no balance error" by showing an alert, which always happen at the first rendering of the component, if balances might not be loaded. To avoid that, we do need to wrap balances into an `O.Option` here.